src/App.jsx

```diff
@@ -3,26 +3,65 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './app/loadTodos';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
   };
 
+  componentDidMount() {
+    getTodos()
+      .then((todosFromServer) => {
+        this.setState({
+          todos: todosFromServer,
+        });
+      });
+  }
+
+  selectUser = (userId) => {
+    this.setState({
+      selectedUserId: userId,
+    });
+  };
+
+  checkHandler = (todoId) => {
+    const todosNew = [...this.state.todos];
+    const todoNew = todosNew.find(todo => todo.id === todoId);
+
+    todoNew.completed = !todoNew.completed;
+    this.setState({ todos: [...todosNew] });
```

Comment:
```suggestion
    this.setState({ todos: todosNew });
```
todosNew is already a copy of todos

Reply:
changed